### ISSUE TITLE
Enable 'xyz is typing..' feature in code editor 

### ISSUE THREAD

**Issue by Lavcodes** [kothariji/SyntaxMeets#35]
**Is your feature request related to a problem? Please describe.**
While multiple users are typing in the code editor at the same time, the other users are not shown. This could create problems when say, two or more of them try to edit at the same place, cannot track progress in real time, who is writing where.
How it looks right now

![syntaxmeets](https://user-images.githubusercontent.com/59924327/110443930-78a55f00-80e2-11eb-92ac-8dbdd81d2f24.png)

**Describe the solution you'd like**

Along with the typing cursor, show the username of who is typing.

**Are you willing to contribute to this issue?** Yes. Would like some direction though. Not entirely sure about how to implement this.

**Describe alternatives you've considered**
Would want it to look something like this.

![Screenshot (16)](https://user-images.githubusercontent.com/59924327/110444815-742d7600-80e3-11eb-9e4b-ac8f5a98395b.png)

**Additional context**

**Comment by Lavcodes** [comment]
I would like to work on this issue as GSSOC'21 participant.

**Comment by kothariji** [comment]
Hi @Lavcodes, First of all, thank you so much for taking out the time to create this issue.

Definitely 🙌, Please go forward to work on this issue. I am assigning this Issue to you 👍
Even I scratched my head a lot to achieve this😆.
Anyways mentors will be guiding us to achieve this feature.
Also to display the name we can use tooltip by material-UI - https://material-ui.com/components/tooltips/



And last but not least,
Keep Contributing 🙏🔥

**Comment by kothariji** [comment]
any updates @Lavcodes ?

**Comment by Lavcodes** [comment]
Yes, I have been looking for some editors that have similar features. 
Here's one open source editor that I found, quite similar to what we're trying to implement here. 
[Link](https://etherpad.org/) to the website. Could you have a look and let me know if similar implementation would be good.

**Comment by kothariji** [comment]
> Yes, I have been looking for some editors that have similar features.
> Here's one open source editor that I found, quite similar to what we're trying to implement here.
> [Link](https://etherpad.org/) to the website. Could you have a look and let me know if similar implementation would be good.

Yes actually we need something like this, and also you can get ideas from google docs, and [Canva](https://www.canva.com/)

But, currently, we are on implementing Redux for SyntaxMeets, so there are a lot of changes in the codebase.
So once its done, you can surely work on this issue. 🙌

**Comment by kothariji** [comment]
@Lavcodes , the REDUX implementation in done. So now we can work on this issue. 
Also I saw [EtherPad](https://etherpad.org/) And yes we can do like that, but also we can think of something similar to [Repl.it](https://replit.com/)

![image](https://user-images.githubusercontent.com/60866749/113862818-87a52d00-97c6-11eb-8615-d58cb008bc0d.png)

Please share your opinions on that 🙌

**Comment by Lavcodes** [comment]
[This](https://replit.com/site/multiplayer) looks great. I was looking for some implementation details on this [blog](https://blog.replit.com/multi) and their github. I was not able to find out implementation code. Could you help with that

**Comment by kothariji** [comment]
> [This](https://replit.com/site/multiplayer) looks great. I was looking for some implementation details on this [blog](https://blog.replit.com/multi) and their github. I was not able to find out implementation code. Could you help with that

Yes, actually we are also doing research on the same. I'll get something, I will definitely share it with you 🙌
